src/NorthwindWebsite/Business/Services/Implementations/ImageCachingService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using NorthwindWebsite.Business.Services.Interfaces;
+using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Constants;
+using NorthwindWebsite.Core.CustomExceptions.BusinessExceptions;
+
+namespace NorthwindWebsite.Business.Services.Implementations;
+
+public class ImageCachingService : IImageCachingService
```

Comment:
Honestly, I would not mark it as service. Why?
Because _Service_ is responsible for domain business operations. In this case, we are not performing business ops, we are just reading/writing it from/to folder to increase performance.
In my opinion, it's better to move this implementation to _Presentation_ layer, because it will be executed only here. Also, we can rename it to `Handler` to make it a little bit more abstract and avoiding binding to BL. Or, if you wish, it can be even placed in _Middleware_ subfolder like `Caching` with all related items: interface, its implementation, Middleware class